drivers/stepper/adi/tmc_spi.h

```diff
@@ -0,0 +1,61 @@
+/**
+ * @file drivers/stepper/adi/stepper.h
+ *
+ * @brief Private API for Trinamic SPI bus
+ *
+ */
+
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024 Carl Zeiss Meditec AG
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_TMC_SPI_H
+#define ZEPHYR_DRIVERS_TMC_SPI_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief TMC SPI INTERFACE
+ * @ingroup io_priv_interfaces
+ * @{
+ *
+ */
+
+#include <zephyr/drivers/spi.h>
+
+/**
+ * @brief Read a register from the TMC module using the SPI Bus.
+ *
+ * @param bus SPI DT information of the bus.
+ * @param register_address Register.
+ * @param data Pointer to read value.
+ *
+ * @return a value from spi_transceive().
+ */
+int tmc_spi_read_register(const struct spi_dt_spec *bus, const uint8_t register_address,
```

Comment:
is this good for all TMC devices? wondering if it should be `tmc5xxx` instead

Reply:
Think, it should work. Ran a check for the `ADDRESS_MASK` on the [Trinamic IC folder](https://github.com/analogdevicesinc/TMC-API/tree/master/tmc/ic): 

```
grep -rInH --include="*.h" ".ADDRESS_MASK" ic
ic/TMC2160/TMC2160_HW_Abstraction.h:18:#define TMC2160_ADDRESS_MASK     0x7F
ic/TMC2300/TMC2300_HW_Abstraction.h:17:#define TMC2300_ADDRESS_MASK     0x7F
ic/TMC5262/TMC5262_HW_Abstraction.h:18:#define TMC5262_ADDRESS_MASK     0x7F
ic/TMC2226/TMC2226_HW_Abstraction.h:18:#define TMC2226_ADDRESS_MASK     			0x7F
ic/TMC7300/TMC7300_HW_Abstraction.h:16:#define TMC7300_ADDRESS_MASK     0x7F
ic/TMC2660/TMC2660_Macros.h:16:#define TMC2660_ADDRESS(datagram) (((datagram) >> TMC2660_ADDRESS_SHIFT) & TMC2660_ADDRESS_MASK)
ic/TMC2660/TMC2660_Constants.h:18:#define TMC2660_ADDRESS_MASK     0x07
ic/TMC5271/TMC5271_HW_Abstraction.h:18:#define TMC5271_ADDRESS_MASK     0x7F
ic/TMC5240/TMC5240_HW_Abstraction.h:17:#define TMC_ADDRESS_MASK         0x7F
ic/TMC5072/TMC5072_HW_Abstraction.h:18:#define TMC5072_ADDRESS_MASK     0x7F
ic/TMC5130/TMC5130_HW_Abstraction.h:16:#define TMC5130_ADDRESS_MASK     0x7F
ic/TMC4671/TMC4671_Fields.h:25:#define TMC4671_CHIP_INFO_ADDRESS_MASK                       0xFF // CHIPINFO_ADDR //
ic/TMC6100/TMC6100_HW_Abstraction.h:16:#define TMC6100_ADDRESS_MASK     0x7F
ic/TMC2130/TMC2130_HW_Abstraction.h:18:#define TMC2130_ADDRESS_MASK     0x7F
ic/TMC4361A/TMC4361A_Constants.h:18:#define TMC4361A_ADDRESS_MASK     TMC_ADDRESS_MASK
ic/TMC2590/TMC2590_Macros.h:18:#define TMC2590_GET_ADDRESS(X) ((uint8_t) ((((X) >> (TMC2590_ADDRESS_SHIFT + 1)) ? ((X) >> TMC2590_ADDRESS_SHIFT) : 0) & TMC2590_ADDRESS_MASK))
ic/TMC2590/TMC2590_Constants.h:18:#define TMC2590_ADDRESS_MASK     0x07
ic/TMC5031/TMC5031_HW_Abstraction.h:17:#define TMC5031_ADDRESS_MASK     0x7F
ic/TMC2225/TMC2225_HW_Abstraction.h:18:#define TMC2225_ADDRESS_MASK     0x7F
ic/TMC5062/TMC5062_HW_Abstraction.h:18:#define TMC5062_ADDRESS_MASK     0x7F
ic/TMC2240/TMC2240_HW_Abstraction.h:17:#define TMC2240_ADDRESS_MASK     0x7F
ic/TMC2224/TMC2224_HW_Abstraction.h:19:#define TMC2224_ADDRESS_MASK     0x7F
ic/TMC5041/TMC5041_HW_Abstraction.h:18:#define TMC5041_ADDRESS_MASK     0x7F
ic/TMC2208/TMC2208_HW_Abstraction.h:18:#define TMC2208_ADDRESS_MASK     0x7F
ic/TMC6200/TMC6200_HW_Abstraction.h:16:#define TMC6200_ADDRESS_MASK     0x7F
ic/TMC2262/TMC2262_HW_Abstraction.h:15:#define TMC2262_ADDRESS_MASK     0x7F
ic/TMC5272/TMC5272_HW_Abstraction.h:18:#define TMC_ADDRESS_MASK         0x7F
ic/TMC2209/TMC2209_HW_Abstraction.h:17:#define TMC_ADDRESS_MASK         0x7F
ic/TMC5160/TMC5160_HW_Abstraction.h:17:#define TMC5160_ADDRESS_MASK     0x7F
```

Reply:
brilliant, still I would rename this to `adi_tmc`

Reply:
yes, the "Trinamic Motion Control" team has been integrated into ADI as "ADI Trinamic", thus "adi_tmc" makes sense. For some new ICs like TMC8100 and TMC9660 the ADDRESS_MASK will be different.

Reply:
Ahh okay, how do we handle this correctly? I'll extend the read write functions to include address masks and then every driver can handle this locally